basic_nodes/src/main/java/club/mondaylunch/gatos/basicnodes/ListDistinctNodeType.java

```diff
@@ -0,0 +1,49 @@
+package club.mondaylunch.gatos.basicnodes;
+
+import java.util.ArrayList;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+
+import club.mondaylunch.gatos.core.data.DataBox;
+import club.mondaylunch.gatos.core.data.DataType;
+import club.mondaylunch.gatos.core.data.ListDataType;
+import club.mondaylunch.gatos.core.graph.connector.NodeConnector;
+import club.mondaylunch.gatos.core.graph.type.NodeType;
+
+public class ListDistinctNodeType extends NodeType.Process {
+    @Override
+    public Map<String, DataBox<?>> settings() {
+        return Map.of();
+    }
+
+    @Override
+    public Set<NodeConnector.Input<?>> inputs(UUID nodeId, Map<String, DataBox<?>> settings, Map<String, DataType<?>> inputTypes) {
+        return Set.of(
+            new NodeConnector.Input<>(nodeId, "input", ListDataType.GENERIC_LIST));
+    }
+
+    @Override
+    public Set<NodeConnector.Output<?>> outputs(UUID nodeId, Map<String, DataBox<?>> settings, Map<String, DataType<?>> inputTypes) {
+        var outputType = inputTypes.getOrDefault("input", ListDataType.GENERIC_LIST);
+        return Set.of(
+            new NodeConnector.Output<>(nodeId, "output", outputType));
+    }
+
+    @Override
+    public Map<String, CompletableFuture<DataBox<?>>> compute(Map<String, DataBox<?>> inputs, Map<String, DataBox<?>> settings, Map<String, DataType<?>> inputTypes) {
+        var inputListCopy = new ArrayList<>(DataBox.get(inputs, "input", ListDataType.GENERIC_LIST).orElseThrow());
+        var outputType = inputTypes.get("input");
+        return Map.of("output", CompletableFuture.completedFuture(
+            this.getGenericListBox(new LinkedHashSet<>(inputListCopy).stream().toList(), outputType))
```

Comment:
To remove the need for the copied list + the hashset, you can just do `DataBox.get(inputs, "input", ListDataType.GENERIC_LIST).orElseThrow().stream().sorted().toList()`

Reply:
this approach does not work as not all types are convertible to Comparable (mainly JsonObject). I assume you mean ".distinct()" instead of ".sorted()"?